src/Illuminate/Routing/ImplicitRouteBinding.php

```diff
@@ -46,7 +46,7 @@ public static function resolveForRoute($container, $route)
                         ? 'resolveSoftDeletableRouteBinding'
                         : 'resolveRouteBinding';
 
-            if ($parent instanceof UrlRoutable && ($route->enforcesScopedBindings() || array_key_exists($parameterName, $route->bindingFields()))) {
+            if ($parent instanceof UrlRoutable && ! $route->preventsScopedBindings() && ($route->enforcesScopedBindings() || array_key_exists($parameterName, $route->bindingFields()))) {
```

Comment:
can't we just add the logic for preventing scopeBinding to `enforcesScopedBinding()`?

From my understanding of below implementation both would never be true the same time, right?

implementation can be almost the same as you have for `preventsScopedBindings()`, only by changing the equality comparison from `false` to `true`

Reply:
> From my understanding of below implementation both would never be true the same time, right?

Don't think so because `enforcesScopedBindings()` is going to return false if `scope_bindings` has not been set (the default), but `array_key_exists($parameterName, $route->bindingFields())` generally returns true so it still evaluates with child binding. 

Are you suggesting something like:
```php
public function enforcesScopedBindings()
{
    return ! isset($this->action['scope_bindings']) || $this->action['scope_bindings'] === true;
}

// ...

if ($parent instanceof UrlRoutable && $route->enforcesScopedBindings() && array_key_exists($parameterName, $route->bindingFields()))) {
```

That might work and allow the removal of `preventsScopedBindings`.

Reply:
You are right. I oversaw this `OR` condition on the `if` clause.

It still feels strange to have both methods that should just be the other's inverse.

Maybe you can propose this new implementation of `Route@enforcesScopedBindings()` :

~~~php
public function enforcesScopedBindings($parameterName = null)
{
    if (isset($this->action['scope_bindings'])) {
        return $this->action['scope_bindings'] === true;
    }
    
    if ($parameterName) {
        return array_key_exists($parameterName, $this->bindingFields());
    }
    
    return false;
}
~~~

Then you can simplify the `if` condition to:

~~~php
if ($parent instanceof UrlRoutable && $route->enforcesScopedBindings($parameterName)) {
~~~

But this would be a breaking change (public method signature/arity changed) and thus would need to be sent to master instead.

Otherwise, I think your current implementation is the best solution -- for adding this feature -- for now.

Reply:
I'm still getting green tests when combining `enforcesScopedBindings()` and `preventsScopedBindings()` as detailed in my previous comment, and it does feel more intuitive to have the single method. 

My concern is that `enforcesScopedBindings()` would now return `true` if `scope_bindings` is not set, whereas previously it would return `false`. 

That won't matter for the single usage in the framework, but i'm not sure whether people would be using this method for some reason in the wild. 

For this reason I think the extra method ensures backwards-compatibility; but happy to refactor to the cleaner version if requested.